pydatastructs/graphs/algorithms.py

```diff
@@ -548,3 +551,84 @@ def strongly_connected_components(graph, algorithm):
         "isn't implemented for finding strongly connected components."
         %(algorithm, graph._impl))
     return getattr(algorithms, func)(graph)
+
+def depth_first_search(
+    graph, source_node, operation, *args, **kwargs):
+    """
+    Implementation of serial depth first search (DFS)
```

Comment:
```suggestion
    Implementation of depth first search (DFS)
```
Parallel version isn't possible for DFS.